File: tests/fixtures/spring/vulnerable/LookupController.java
import java.sql.*;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class LookupController {
    private final Connection connection;
    public LookupController(Connection connection) { this.connection = connection; }

    @GetMapping("/lookup")
    public ResultSet lookup(@RequestParam String name) throws SQLException {
        return connection.createStatement().executeQuery("SELECT * FROM users WHERE name='" + name + "'");
    }
}

File: tests/fixtures/csharp-mvc/fixed/LookupController.cs
using System.Data.Common;
using System.Web.Mvc;

[RoutePrefix("users")]
public class LookupController : Controller {
    private readonly DbConnection connection;
    public LookupController(DbConnection connection) { this.connection = connection; }

    [HttpGet]
    [Route("")]
    public DbDataReader Lookup(string name) {
        var command = connection.CreateCommand();
        command.CommandText = "SELECT * FROM users WHERE name=@name";
        var parameter = command.CreateParameter();
        parameter.ParameterName = "@name";
        parameter.Value = name;
        command.Parameters.Add(parameter);
        return command.ExecuteReader();
    }
}
